lib/file-source/src/paths_provider/glob.rs

```diff
@@ -1,11 +1,38 @@
 //! [`Glob`] paths provider.
 
 use super::PathsProvider;
-
 use glob::Pattern;
+use globwalk::glob;
 use std::path::PathBuf;
 
-pub use glob::MatchOptions;
+#[derive(Debug, snafu::Snafu)]
+/// An error that arised either during parsing or execution of this glob.
+pub enum GlobError {
+    /// Include glob pattern could not be parsed.
+    #[snafu(display("Include glob pattern {} could not be parsed: {}", glob, error))]
+    InvalidIncludePattern {
+        /// The glob string that produced the error.
+        glob: String,
+        /// The underlying error.
+        error: globwalk::GlobError,
```

Comment:
This and the rest of `error` variables in this enum can be renamed to `source`. With that snafu's `context` and `with_context` methods can be used with `use snafu::ResultExt;`

Reply:
Aha! That's quite handy.

Reply:
Thanks for the tip 👍